Fix: Update system bars theme based on app theme

yusufnasserdev opened the pull request:
# What does this PR do?

This PR introduces a new composable function `UpdateSystemBarsTheme` which updates the appearance of the status and navigation bars to match the current app theme (light or dark).

The main activity now observes the app theme preference and system theme to determine whether to use a dark theme for the `NovixTheme`. This determination is then passed to `UpdateSystemBarsTheme` to ensure the system UI elements are styled consistently.

## Screenrecords

**_Check the system top status bar_**.

Before Vs After  

[screencast-Genymotion-2025-08-07_13.01.25.521.webm](https://github.com/user-attachments/assets/66a69d1a-da5b-4f85-b95e-1a7368bdc559) 

[screencast-Genymotion-2025-08-07_13.02.03.142.webm](https://github.com/user-attachments/assets/bd2f48d6-84a3-461f-890b-267ceb74682c)

yusufnasserdev committed:
Refactor: update system bars theme based on app theme

london-bot[bot] committed:
Branch was auto-updated.

yusufnasserdev committed:
Merge branch 'develop' into hotfix/system-bars-theme

# Conflicts:
#	app/src/main/java/com/london/app/MainActivity.kt

ahmeddsamii reviewed (dismissed):
the application is switching based on mobile system

yusufnasserdev committed:
refactor: remove system theme and set dark theme by default

yusufnasserdev committed:
refactor: update system bar theme logic

- Makes the system bars transparent, except on 3-button navigation mode where the status bar gets a translucent scrim.
- The colors of the system icons and the scrim are adjusted based on the system light or dark theme.
- Updates the status bar and navigation bar icon colors based on the current theme.

yusufnasserdev committed:
Merge branch 'develop' into hotfix/system-bars-theme

london-bot[bot] committed:
Branch was auto-updated.

yusufnasserdev committed:
feat: add isDarkTheme property to color schemes

yusufnasserdev committed:
refactor: remove AppTheme enum and integrate isDarkTheme into NovixColors

yusufnasserdev committed:
refactor: remove LocalAppTheme from NovixTheme

yusufnasserdev committed:
Fix: Change translucent scrim color to transparent

yusufnasserdev committed:
feat: initialize app theme in AccountViewModel

yusufnasserdev committed:
fix: update DefaultTopBar icon based on NovixTheme colors

The `DefaultTopBar` icon selection logic was updated to use `NovixTheme.colors.isDarkTheme` instead of `NovixTheme.theme.isDark()`. This ensures the correct Novix logo (light or dark) is displayed based on the current theme's color scheme.

london-bot[bot] committed:
Branch was auto-updated.

Mostafa-alsaygh commented on app/src/main/java/com/london/app/MainActivity.kt:
```diff
@@ -49,3 +51,27 @@ class MainActivity : ComponentActivity() {
     }
 }
+
+@Suppress("DEPRECATION")
```
is there another way to not use deprecation thing

Mostafa-alsaygh replied:
search for it

yusufnasserdev replied:
There is  not, we're trying to serve all android versions, we had to use this.

It's only deprecated beyond Android 35.

Mostafa-alsaygh commented on designSystem/src/main/java/com/london/designsystem/color/DarkThemeColors.kt:
```diff
@@ -56,4 +56,5 @@ val DarkNovixColors = NovixColors(
     linearGradient = LinearGradientDark,
     blackLinearGradient = BlackLinearGradientDark,
     whiteLinearGradient = WhiteLinearGradientDark,
+    isDarkTheme = true
```
that's not a correct place for it, only a colors is here

Mostafa-alsaygh commented on designSystem/src/main/java/com/london/designsystem/color/LightThemeColors.kt:
```diff
@@ -56,4 +56,5 @@ val LightNovixColors = NovixColors(
     linearGradient = LinearGradientLight,
     blackLinearGradient = BlackLinearGradientLight,
     whiteLinearGradient = WhiteLinearGradientLight,
+    isDarkTheme = false
```
that's not a correct place for it, only a colors is here

Mostafa-alsaygh commented on designSystem/src/main/java/com/london/designsystem/color/NovixColor.kt:
```diff
@@ -29,4 +29,5 @@ data class NovixColors(
     val linearGradient: Color,
     val blackLinearGradient: Color,
     val whiteLinearGradient: Color,
+    val isDarkTheme: Boolean,
```
that's not a correct place for it, only a colors is here

london-bot[bot] committed:
Branch was auto-updated.

yusufnasserdev committed:
Merge branch 'develop' into hotfix/system-bars-theme

# Conflicts:
#	app/src/main/java/com/london/app/MainActivity.kt
#	data/src/main/java/com/london/data/local/preference/AppPreferencesServiceImpl.kt
#	designSystem/src/main/java/com/london/designsystem/theme/NovixTheme.kt
#	designSystem/src/main/java/com/london/designsystem/theme/Theme.kt
#	domain/src/main/java/com/london/domain/theme/AppTheme.kt
#	presentation/src/main/java/com/london/presentation/feature/home/HomeScreen.kt

yusufnasserdev committed:
fix: update app theme initialization

london-bot[bot] committed:
Branch was auto-updated.
